gradio/routes.py

```diff
@@ -575,6 +574,10 @@ async def join_queue(
                 )
                 set_task_name(task, event.session_hash, event.fn_index, batch=False)
             else:
+                if blocks._queue.event_queue.__len__() > 0:
```

Comment:
@abidlabs Not sure if this is the right way to go about "ignoring" events from the queue

Reply:
Taking a look

Reply:
I left some comments on the frontend parts. I need to understand those better before I can comment here. But my feeling is that you shoudn't even need to make any changes to the backend if the frontend changes are correct.

If `trigger_mode` is `always_last`, then in the frontend, if an event is pending, you'd just stack the incoming requests, and only send to the backend the final one (so that'd be the only one added to the queue)